ready for comps

thetaspirit opened the pull request:
note that there are a _few_ features that have not been tested, but most of them are presets

this branch can entirely replace master.  when you merge, it probably will

it's game time

thetaspirit committed:
enabled swerve control

thetaspirit committed:
updated wheel locations for drive

thetaspirit committed:
Merge branch 'master' of https://github.com/SaintsRobotics/ShoppingKart2019

thetaspirit committed:
comp configs are set to the same as test configsas in, comp configs are inaccurateRobot.java has the dimensions of the comp bot

thetaspirit committed:
looks like these never made it to the last push

which makes sense because the previous commit says the comp configs are
inaccurate lol

thetaspirit committed:
comp bot configs have been tested w/ drive practice

arms and heading pids re-tuned
arms target angles set

thetaspirit committed:
added task to reset arms encoder offset

Task has not been tested
the fullInPosition is zero
and the other positions are set relative to that

thetaspirit committed:
push for remote work

plan to:
merge the package organization branch with this one
add other arms encoder reset task
double check some more stuff

thetaspirit committed:
revised resetting the arms enocder

has not been calibrated or tested

thetaspirit committed:
arms reset works, has been teseted

thetaspirit committed:
edited arm offsets

honestly, this stuff will probalby need retuning again soon

thetaspirit committed:
changed a swerve wheel offset cuz 'twas replaced

calibrated arms (again)

dpk2442 committed:
Merge branch 'master' into comp-bot-configs

thetaspirit committed:
renamed motor and sensor config classes

Opportunitree is the comp bot
Curiositree is the test bot

thetaspirit committed:
added a few more robot headings

thetaspirit committed:
if the left drive trigger is held, the robot goes straight fwd

thetaspirit committed:
decoupled ArmsControl and ArmsInput

added DetatchPannel task
has not been tested

thetaspirit committed:
Merge branch 'comp-bot-configs' of https://github.com/SaintsRobotics/ShoppingKart2019 into comp-bot-configs

thetaspirit committed:
Merge branch 'master' of https://github.com/SaintsRobotics/ShoppingKart2019

dpk2442 committed:
refactor robot init to use config file and remove Robot singleton

Summary of the changes:
    - robotInit now loads a config file first. If a jumper is
      present on the specified digital input, the test config
      will be loaded.
    - Motors and Sensors are no longer abstract, and initialize
      all motors using the values from the passed in config.
    - TurnConfiguration has been deleted and replaced with PidConfig
    - All tasks using PID controllers were updated to take a
      PidConfig object as an argument.
    - Robot.instance was removed, and all tasks were updated to
      take all needed inputs in the constructors.

thetaspirit committed:
Merge branch 'comp-bot-configs' into curiositree-testing

thetaspirit committed:
updated curiositree wheel offsets

thetaspirit committed:
calibrated arms

thetaspirit committed:
lift tuning and calibration

thetaspirit committed:
added more fields in properties

implemented the robot.location fields in robot.java

thetaspirit committed:
WIP in the middle of debugging merge

thetaspirit committed:
still need to implement non-pid magic nums

works for drive practice

thetaspirit committed:
added magic numbers to settings along with pid values

Commit:
fixed bug in Settings.buildLoc(Settings.java:52)

thetaspirit committed:
fixed config-related bugs

deploys and drives/operates

thetaspirit committed:
vision successfully integrated

captainrex493 committed:
Changes to using custom operator board with led buttons

thetaspirit committed:
docking stops after outputs are negligible for 100 frames

thetaspirit committed:
Merge branch 'config-test-merge' of https://github.com/SaintsRobotics/ShoppingKart2019 into config-test-merge

thetaspirit committed:
operator board successfully integrated

thetaspirit committed:
added vision "pipeline" file

captainrex493 committed:
Adds boolean to switch between custom control board and xbox controller

thetaspirit committed:
added more setting fields, set up for comp configs

thetaspirit committed:
re-enabled drive straight fwd feature

captainrex493 committed:
test bot configured and tuned

thetaspirit committed:
comp configs and settings updated

arms position upon deploy is now customizable
using an xbox controller with opperator board has a few extra features
DetatchPannel has not officially been tested (or tuned)
hatch heights 2 and 3 have not been calibrated

captainrex493 committed:
updated to wpilib 2019.4.1

thetaspirit committed:
updated to wpilib 2019.4.1

thetaspirit committed:
Merge remote-tracking branch 'origin/youCanLiteralyNameItAnythingBecauseIHopetoDeleteItTonight' into config-test-merge

captainrex493 reviewed (dismissed):
Looks good to me. Maybe have one more person look through this because it is very large.

thetaspirit commented:
There are a lot of commits for this branch.  Would it be a good idea to have multiple commits?  The easiest thing to do would be to rebase, but this commit history here is a bit messy.

thetaspirit committed:
drive control update

reacrtivated boost (rightTrigger() > .5)
changed drive and turn gain to .75

thetaspirit committed:
Auburn Mountainview post competition patch

Lift throttling
kicker backpass (commented out)
Fixed competition presets

thetaspirit committed:
Auburn Mountainview Patch

Patch Harder

thetaspirit committed:
Auburn Mountainview Patch

Patch Harder

captainrex493 committed:
tuned docking

added feature:
right trigger makes robot drive straight backwards

captainrex493 committed:
lift speed throttling is now in lift input

captainrex493 committed:
added try catch (hasn't been tested)

captainrex493 committed:
added second dock pid config, tuned various values

thetaspirit committed:
Glacier peak (#13)

* calibrated vision trgets to comp field

* lowered lift throttle, recalibrated arms, no change in Robot besides formatting

* decreased arm pid output range

* decreased heading pid output range, lowered turn gain

* lowered max lift height with encoder

* decreased distance output range

* added boost to full speed for drive (left bumper)

* added lower limit to lift

if the encoder ever reads less than 17 inches, the lift is stopped

* update lift upper limit

lift can go about eight inches higher now

* calibrated lift presets

Glacier Peak (practice field)
and calibrated arms
